File: problem2.java
/**
*@Author Daniel Gravel
*Calculates the sum of all even fibinacci numbers up to 4 million
*/
public class problem2{
	public int evenFibonacci(){
		int first = 1;
		int second = 0;
		int sum = 0;

		while(first <= 4000000){
			if(first%2 == 0){
				sum += first;
			}
			int temp = first + second;
			second = first;
			first = temp;
		}
		return sum;
	}

	public static void main(String args[]){
		System.out.println(new problem2().evenFibonacci() + "");
	}
}

File: python/problem2.py
"""
Finds the sum of all even fibonacci numbers up to the given limit
@author: Gravel
"""
def problem2(limit):
    sum = 0
    a,b = 1, 1
    while a <= limit:
        if a % 2 == 0:
            sum += a
        a, b = b, a+b
    print(sum)
    return(sum)
        
